src/v/cluster/rm_stm.cc

```diff
@@ -2529,8 +2521,8 @@ rm_stm::apply_snapshot(stm_snapshot_header hdr, iobuf&& tx_ss_buf) {
         const auto pid = entry.pid;
         auto it = _log_state.seq_table.find(pid);
         if (it == _log_state.seq_table.end()) {
-            seq_entry_wrapper seq{std::move(entry)};
-            _log_state.seq_table.try_emplace(it, pid, std::move(seq));
+            _log_state.seq_table.try_emplace(
+              it, pid, seq_entry_wrapper{.entry = std::move(entry)});
```

Comment:
this looks fine, but it also looks like the before/after have identical behavior. is there something important going on here?

Reply:
no, not really. this is a revert commit of the original (incorrect) fix that gets it back to origin/dev state.